### PR TITLE
[fix](Nereids) support string cast to complex type

### PR DESCRIPTION
**Pull request by morrySnow**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morrySnow** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/check/CheckCast.java [hunk 1]
```diff
@@ -57,6 +58,9 @@ private boolean check(DataType originalType, DataType targetType) {
         if (originalType.equals(targetType)) {
             return true;
         }
+        if (originalType instanceof CharacterType && !(targetType instanceof PrimitiveType)) {
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/check/CheckCast.java:61]: line 84 can be removed.

**Comment by morrySnow** [comment]
run buildall

**Review by englefly — DISMISSED** [review]

**Comment by morrySnow** [comment]
run buildall

**Review by starocean999 — DISMISSED** [review]

**Comment by morrySnow** [comment]
run buildall

**Review by englefly — APPROVED** [review]

**Review by starocean999 — APPROVED** [review]

**Pull request merged**